Mathlib/SetTheory/Cardinal/Aleph.lean

```diff
@@ -91,8 +96,152 @@ def isInitialIso : {x // IsInitial x} ≃o Cardinal where
   right_inv x := card_ord x
   map_rel_iff' {a _} := a.2.card_le_card
 
--- TODO: define `omega` as the enumerator function of `IsInitial`, and redefine
--- `aleph x = (omega x).card`.
+/-- The `omega'` function gives the initial ordinals listed by their ordinal index. `omega' n = n`,
+`omega' ω = ω`, `aleph' (ω + 1) = ω₁`, etc.
+
+For the more common omega function skipping over finite ordinals, see `Ordinal.omega`. -/
+def omega' : Ordinal.{u} ↪o Ordinal.{u} where
+  toFun := enumOrd {x | IsInitial x}
+  inj' _ _ h := enumOrd_injective not_bddAbove_isInitial h
+  map_rel_iff' := enumOrd_le_enumOrd not_bddAbove_isInitial
+
+theorem coe_omega' : omega' = enumOrd {x | IsInitial x} :=
+  rfl
+
+theorem omega'_strictMono : StrictMono omega' :=
+  omega'.strictMono
+
+theorem omega'_lt {o₁ o₂ : Ordinal} : omega' o₁ < omega' o₂ ↔ o₁ < o₂ :=
+  omega'.lt_iff_lt
+
+theorem omega'_le {o₁ o₂ : Ordinal} : omega' o₁ ≤ omega' o₂ ↔ o₁ ≤ o₂ :=
+  omega'.le_iff_le
+
+theorem omega'_max (o₁ o₂ : Ordinal) : omega' (max o₁ o₂) = max (omega' o₁) (omega' o₂) :=
+  omega'.monotone.map_max
+
+theorem isInitial_omega' (o : Ordinal) : IsInitial (omega' o) :=
+  enumOrd_mem not_bddAbove_isInitial o
+
+theorem le_omega'_self (o : Ordinal) : o ≤ omega' o :=
+  omega'_strictMono.le_apply
+
+@[simp]
+theorem omega'_zero : omega' 0 = 0 := by
+  rw [coe_omega', enumOrd_zero]
+  exact csInf_eq_bot_of_bot_mem isInitial_zero
+
+@[simp]
+theorem omega'_natCast (n : ℕ) : omega' n = n := by
```

Comment:
Can we also have `omega' (ofNat n) = n`?

Reply:
Sure, but I still don't get why we often need `natCast` and `ofNat` versions of the same theorems. What's up with that?

Reply:
`(n : Ordinal)` is `Nat.cast n`. `(5 : Ordinal)` is `OfNat.ofNat 5`